lib/rules/no-script-url.js

```diff
@@ -31,18 +33,26 @@ module.exports = {
 
     create(context) {
 
+        /**
+         * Check whether a node's static value starts with "javascript:" or not.
+         * And report an error for unexpected script URL.
+         * @param {ASTNode} node node to check
+         * @returns {void}
+         */
+        function check(node) {
+            const value = astUtils.getStaticStringValue(node);
+
+            if (typeof value === "string" && value.toLowerCase().indexOf("javascript:") === 0) {
+                context.report({ node, messageId: "unexpectedScriptURL" });
+            }
+        }
         return {
-
-            Literal(node) {
-                if (node.value && typeof node.value === "string") {
-                    const value = node.value.toLowerCase();
-
-                    if (value.indexOf("javascript:") === 0) {
-                        context.report({ node, messageId: "unexpectedScriptURL" });
-                    }
+            Literal: check,
```

Comment:
The old version used to check if it is a string literal, but the new version checks `String()` value of all literals.

While it's now impossible that any other literal's `String()` value starts with `javascript:`, it might be more correct and future-proof to still explicitly check if it is a string literal.